doc/company.texi

```diff
@@ -0,0 +1,1499 @@
+Select the previous candidate
+(@code{company-complete-previous-or-abort}).
+
+@item RET
+@itemx <return>
+@kindex RET
+@cindex complete
+@findex company-complete-selection
+Insert the selected candidate (@code{company-complete-selection}).
+
+@item TAB
+@itemx <tab>
+@kindex TAB
+@cindex common part
+@findex company-complete-common
+Insert the common part of all the candidates
+(@code{company-complete-common}).
+
+@item C-g
+@itemx <ESC ESC ESC>
+@kindex C-g
+@cindex abort
+@cindex quit
+@cindex finish
+@cindex stop
+@cindex cancel
+@findex company-abort
+Cancel @emph{company-mode} activity (@code{company-abort}).
+
+@item C-h
+@itemx <f1>
+@kindex C-h
+@cindex doc
+@findex company-show-doc-buffer
+Display a buffer with the documentation for the selected candidate
+(@w{@code{company-show-doc-buffer}}).
+
+@item C-w
+@kindex C-w
+@cindex definition
+@cindex location
+@findex company-show-location
+Display a buffer with the definition of the selected candidate
+(@w{@code{company-show-location}}).
+
+@end table
+
+The full list of the default key bindings is stored in the variables
+@code{company-active-map} and @code{company-search-map} @footnote{For
+a more user-friendly output of the pre-defined key bindings, utilize
+@w{@kbd{M-x describe-keymap @key{RET} company-active-map}} or @w{@kbd{C-h
+f @key{RET} company-mode}}.}.
+
+Moreover, Company is bundled with a number of convenience commands
+that do not have default key bindings defined.  The following examples
+illustrate how to assign key bindings to such commands.
+
+@lisp
+(global-set-key (kbd "<tab>") #'company-indent-or-complete-common)
+@end lisp
+
+@lisp
+(with-eval-after-load 'company
+  (define-key company-active-map (kbd "M-/") #'company-complete))
+@end lisp
+
+@lisp
+(with-eval-after-load 'company
+  (define-key company-active-map
+              (kbd "TAB")
+              #'company-complete-common-or-cycle)
+  (define-key company-active-map
+              (kbd "<backtab>")
+              (lambda ()
+                (interactive)
+                (company-complete-common-or-cycle -1))))
+@end lisp
+
+In the same manner, an additional key can be assigned to a command or
+a command can be unbound from a key.  For instance:
+
+@lisp
+(with-eval-after-load 'company
+  (define-key company-active-map (kbd "M-.") #'company-show-location)
+  (define-key company-active-map (kbd "RET") nil))
+@end lisp
+
+@c TODO:
+@c Maybe list some of the others, not mentioned yet, ``core'' commands:
+@c company-manual-begin
+@c company-select-next
+@c company-select-previous
+@c company-select-first
+@c company-select-last
+@c company-next-page
+@c company-previous-page
+@c company-select-next-if-tooltip-visible-or-complete-selection
+@c company-version
+@c But that should not confuse a newbie with too much information.
+@c Or maybe describe how to list all the available commands
+@c in a simple way (would it be M-x occur <interactive> ?).
+@c Might be provide a separate reference with Company commands
```

Comment:
If these don't seem essential to you, maybe indeed avoid documenting them specifically.

And while `company-select-next` is indeed important, you probably indirectly described it already if you mentioned when `C-n` does when the completion is in progress.

Reply:
Okay, then I'm removing these comments for now. These descriptions can be added later, if there's such a need.

> while company-select-next is indeed important, you probably indirectly described it already if you mentioned when C-n

Yeah, almost. `C-n` is documented with `company-select-next-or-abort`. Maybe `company-select-next` can be mentioned there too, though not fully sure. I think this also can be resolved on a later stage.

Reply:
Sounds good.